examples/createLayer_custom_UI_functions.html

```diff
@@ -0,0 +1,293 @@
+        </div>
+    </script>
+
+<script>
+  function main() {
+    
+    var lat;
+    var lon;
+
+    // define map object
+    var map = new L.Map('map', {
+      zoomControl: false,
+      center: [43, 0],
+      zoom: 3
+
+    });
+  
+    /* Define CartoDB layer with createLayer(). More information here: 
+
+    http://docs.cartodb.com/cartodb-platform/cartodb-js/api-methods/#cartodbcreatelayermap-layersource--options--callback */
+    cartodb.createLayer(map, {
+      user_name: 'oboix',
+      type: 'cartodb',
+      sublayers: [
+        {
+          type: "http",
+          urlTemplate: "http://{s}.basemaps.cartocdn.com/dark_nolabels/{z}/{x}/{y}.png",
+          subdomains: [ "a", "b", "c" ]
+        },
+        {
+           type: "mapnik",
+           sql: 'select * from populated_places_spf',
+           cartocss: '#populated_places_spf[adm0name = "Spain"]{ marker-fill: #fbb4ae; marker-allow-overlap: true;}#populated_places_spf[adm0name = "Portugal"]{ marker-fill: #ccebc5; marker-allow-overlap: true;}#populated_places_spf[adm0name = "France"]{ marker-fill: #b3cde3; marker-allow-overlap: true;}',
+           interactivity: ['cartodb_id','adm0name','name','pop_max']
+         
+        },
+        {
+          type: "http",
+          urlTemplate: "http://{s}.basemaps.cartocdn.com/dark_only_labels/{z}/{x}/{y}.png",
+          subdomains: [ "a", "b", "c" ]
+        }
+      ]
+    })
+  .addTo(map) // add cartodb layer and basemap to map object
+  .done(function(layer) {
+
+    /*Once that the basemap and cartodb layer are added....*/
+      
+        /* Enable mouse events with CartoDB layers */
+
+        // 1- Set CartoDB layer interaction
+        layer.setInteraction(true);
+
+        // 2- Custom mouseover event on CartoDB layers
+        layer.on('featureOver',function(e,latlng,pos,data){
+          lat = (latlng[0]).toFixed(2) // show latitude of clicked point in the console
+          lon = (latlng[1]).toFixed(2) // show longitude of clicked point in the console
+          
+        // 3- Add coordinates of the selected geometry in the div element 
+        //  with id = "box" 
+           document.getElementById("lat").innerHTML = lat;
+           document.getElementById("lon").innerHTML = lon;
+        });
+
+      /* Set custom infowindow (= click infowindow) More information here:
+      http://docs.cartodb.com/cartodb-platform/cartodb-js/ui-functions/#cartodbvisvisaddinfowindowmap-layer-fields--options
+
+      */
+
+       cdb.vis.Vis.addInfowindow(
+          map, layer, ['adm0name','name','pop_max'],
+          {
+             infowindowTemplate: $('#infowindow_template').html()
+          });
+
+       /* Set custom tooltip (= hover infowindow) More information here:
+            http://docs.cartodb.com/cartodb-platform/cartodb-js/ui-functions/#visaddoverlaytooltip
+
+         */
+
+        // 1- Define tootltip with createLayer
+         var tooltip = layer.leafletMap.viz.addOverlay({
```

Comment:
This is a bit hacky (`layer.leafletMap.viz`). There isn't a better way of doing it sadly, but do we wan't to encourage users to use this (given that `layer.leafletMap.viz` is undocumented) ?